packages/beacon-state-transition/src/altair/epoch/processInactivityUpdates.ts

```diff
@@ -16,16 +16,21 @@ export function processInactivityUpdates(state: CachedBeaconState<altair.BeaconS
   const {config} = state;
   const {INACTIVITY_SCORE_BIAS, INACTIVITY_SCORE_RECOVERY_RATE} = config;
   const inActivityLeak = isInInactivityLeak((state as unknown) as phase0.BeaconState);
+  const {inactivityScores} = state;
   for (let i = 0; i < process.statuses.length; i++) {
     const status = process.statuses[i];
     if (hasMarkers(status.flags, FLAG_ELIGIBLE_ATTESTER)) {
+      let inactivityscore = inactivityScores[i];
       if (hasMarkers(status.flags, FLAG_PREV_TARGET_ATTESTER_OR_UNSLASHED)) {
-        state.inactivityScores[i] -= Math.min(1, state.inactivityScores[i]);
+        inactivityscore -= Math.min(1, inactivityscore);
       } else {
-        state.inactivityScores[i] += Number(INACTIVITY_SCORE_BIAS);
+        inactivityscore += Number(INACTIVITY_SCORE_BIAS);
       }
       if (!inActivityLeak) {
-        state.inactivityScores[i] -= Math.min(Number(INACTIVITY_SCORE_RECOVERY_RATE), state.inactivityScores[i]);
+        inactivityscore -= Math.min(Number(INACTIVITY_SCORE_RECOVERY_RATE), inactivityscore);
+      }
+      if (inactivityscore !== state.inactivityScores[i]) {
```

Comment:
Instead of doing a second state read you could keep the previous value scoped here like `inactivityscorePrev` then
```ts
if (inactivityscore !== inactivityscorePrev) ...
```